waiter/src/waiter/core.clj

```diff
@@ -533,11 +533,12 @@
                           (utils/create-component entitlement-config))
    :fallback-state-atom (pc/fnk [] (atom {:available-service-ids #{}
                                           :healthy-service-ids #{}}))
-   :http-client (pc/fnk [[:settings [:instance-request-properties connection-timeout-ms]]
-                         server-name]
-                  (http-utils/http-client-factory {:conn-timeout connection-timeout-ms
-                                                   :follow-redirects? false
-                                                   :user-agent server-name}))
+   :http-clients (pc/fnk [[:settings [:instance-request-properties connection-timeout-ms]]
+                          server-name]
+                   (http-utils/prepare-http-clients
```

Comment:
We now return a dictionary of http clients and the appropriate client (HTTP/1 or HTTP/2 complaint) is chosen while making requests.